futures-util/src/sink/mod.rs

```diff
@@ -250,8 +251,11 @@ pub trait SinkExt<Item>: Sink<Item> {
     fn send_all<'a, St>(&'a mut self, stream: &'a mut St) -> SendAll<'a, Self, St>
     where
         St: TryStream<Ok = Item, Error = Self::Error> + Stream + Unpin + ?Sized,
+        // St: Stream<Item = Result<Item, Self::Error>> + Unpin + ?Sized,
         Self: Unpin,
     {
+        // TODO: type mismatch resolving `<St as Stream>::Item == std::result::Result<Item, <Self as futures_sink::Sink<Item>>::Error>`
+        // assert_future::<Result<(), Self::Error>, _>(SendAll::new(self, stream))
```

Comment:
IIUC, this is not because the bounds are not satisfied, but because the compiler cannot resolve projection (assoc type).

Reply:
Since `TryStream` is sealed, I think it's safe to replace the `TryStream` bounds with `Stream<Item = Result<..>>` bounds, but for now, I'll leave it as it is.